### ISSUE TITLE
chef-shell breaks when run in a shell from `sudo`

### ISSUE THREAD

**Issue by jaymzh** [chef/chef#10747]
## Description

Chef-shell doesn't work in standalone mode

## Chef Version
16.7.61

## Platform Version
Ubuntu 18.04

## Replication Case
`chef-shell -a`

## Client Output

```
$ sudo su -
# chef-shell -a
Redirecting to cinc-shell
loading configuration: none (standalone session)
true
Switch to inspect mode.
Session type: standalone
Loading..........done.

Welcome to the cinc-shell 16.7.61
For usage see https://docs.chef.io/chef_shell/

run `help' for help, `exit' or ^D to quit.

cinc (16.7.61)> /opt/cinc/embedded/lib/ruby/2.7.0/reline/ansi.rb:74:in `raw': Inappropriate ioctl for device (Errno::ENOTTY)
	from /opt/cinc/embedded/lib/ruby/2.7.0/reline/ansi.rb:74:in `getc'
	from /opt/cinc/embedded/lib/ruby/2.7.0/reline.rb:271:in `block in read_io'
	from /opt/cinc/embedded/lib/ruby/2.7.0/reline.rb:270:in `loop'
	from /opt/cinc/embedded/lib/ruby/2.7.0/reline.rb:270:in `read_io'
	from /opt/cinc/embedded/lib/ruby/2.7.0/reline.rb:239:in `block in inner_readline'
	from /opt/cinc/embedded/lib/ruby/2.7.0/reline.rb:238:in `loop'
	from /opt/cinc/embedded/lib/ruby/2.7.0/reline.rb:238:in `inner_readline'
	from /opt/cinc/embedded/lib/ruby/2.7.0/reline.rb:175:in `readmultiline'
	from /opt/cinc/embedded/lib/ruby/2.7.0/forwardable.rb:235:in `readmultiline'
	from /opt/cinc/embedded/lib/ruby/2.7.0/forwardable.rb:235:in `readmultiline'
	from /opt/cinc/embedded/lib/ruby/2.7.0/irb/input-method.rb:294:in `gets'
	from /opt/cinc/embedded/lib/ruby/2.7.0/irb.rb:519:in `block (2 levels) in eval_input'
	from /opt/cinc/embedded/lib/ruby/2.7.0/irb.rb:704:in `signal_status'
	from /opt/cinc/embedded/lib/ruby/2.7.0/irb.rb:518:in `block in eval_input'
	from /opt/cinc/embedded/lib/ruby/2.7.0/irb/ruby-lex.rb:182:in `lex'
	from /opt/cinc/embedded/lib/ruby/2.7.0/irb/ruby-lex.rb:154:in `block (2 levels) in each_top_level_statement'
	from /opt/cinc/embedded/lib/ruby/2.7.0/irb/ruby-lex.rb:151:in `loop'
	from /opt/cinc/embedded/lib/ruby/2.7.0/irb/ruby-lex.rb:151:in `block in each_top_level_statement'
	from /opt/cinc/embedded/lib/ruby/2.7.0/irb/ruby-lex.rb:150:in `catch'
	from /opt/cinc/embedded/lib/ruby/2.7.0/irb/ruby-lex.rb:150:in `each_top_level_statement'
	from /opt/cinc/embedded/lib/ruby/2.7.0/irb.rb:537:in `eval_input'
	from /opt/cinc/embedded/lib/ruby/gems/2.7.0/gems/chef-16.7.61/lib/chef/shell.rb:92:in `block in start'
	from /opt/cinc/embedded/lib/ruby/gems/2.7.0/gems/chef-16.7.61/lib/chef/shell.rb:91:in `catch'
	from /opt/cinc/embedded/lib/ruby/gems/2.7.0/gems/chef-16.7.61/lib/chef/shell.rb:91:in `start'
	from /opt/cinc/embedded/lib/ruby/gems/2.7.0/gems/chef-bin-16.7.61/bin/cinc-shell:31:in `<top (required)>'
	from /opt/cinc/bin/cinc-shell:152:in `load'

```

**Comment by jaymzh** [comment]
More fun...
I can repro this with `sudo su -`, `sudo -s`, `sudo -i`.
@lamont-granquist on the other hand can't, both on Ubuntu 18.04.
And I tried taking his exact `/etc/sudoers`, but still no go.

**Comment by ramereth** [comment]
@jaymzh just to be sure, does this also happen with Chef? I see you're using Cinc there..

**Comment by jaymzh** [comment]
@ramereth it's cinc, but it's the wrapper, not the build. Filed https://gitlab.com/cinc-project/distribution/client/-/issues/30

tl;dr for people who stumble into this - until the above is fixed call `cinc-shell` directly when using cinc.
